main.go

```
@@ -0,0 +1,227 @@
+
+	// Thresholds
+	cpuWarn := flag.Float64("cpu-warning", 50, "CPU usage warning threshold")
+	cpuCrit := flag.Float64("cpu-critical", 90, "CPU usage critical threshold")
+	memWarn := flag.Float64("mem-warning", 50, "RAM usage warning threshold")
+	memCrit := flag.Float64("mem-critical", 90, "RAM usage critical threshold")
+	fanWarn := flag.Float64("fan-warning", 3000, "Fan speed warning threshold")
+	tempWarn := flag.Float64("temp-warning", 50, "Temperature warning threshold")
+	tempCrit := flag.Float64("temp-critical", 70, "Temperature critical threshold")
+	statsWarn := flag.Float64("stats-warning", 5, "Port stats warning threshold")
+	statsCrit := flag.Float64("stats-critical", 20, "Port stats critical threshold")
+
+	var portsToCheck intSliceFlag = []int{1, 2, 3, 4, 5, 6, 7, 8}
+	flag.Var(&portsToCheck, "port", "Ports to check (repeatable)")
+
+	help := flag.Bool("help", false, "Show this help")
+	flag.BoolVar(help, "h", false, "Show this help (shorthand)")
+
+	flag.Parse()
+
+	if *help || *username == "" || *password == "" {
+		flag.Usage()
+		return
+	}
+
+	n, err := netgear.NewNetgear(*baseURL, *username, *password)
+	if err != nil {
+		fmt.Printf("URL error: %v", err)
+		os.Exit(check.Unknown)
+	}
+	if err := n.Login(); err != nil {
+		fmt.Printf("Error while trying to login: %v\n", err)
+		os.Exit(int(check.Unknown))
+	}
+	defer func() { _ = n.Logout() }()
+
+	if slices.Contains(mode, "all") {
+		mode = append(mode, "basic", "ports", "poe")
+	}
+
+	worstStatus := check.OK
+
+	// Basic check
+	if slices.Contains(mode, "basic") {
+		deviceInfo, err := n.DeviceInfo()
+		if err != nil {
+			fmt.Printf("Error retrieving device info: %v\n", err)
+			os.Exit(int(check.Unknown))
+		}
+
+		upTime := deviceInfo.DeviceInfo.Details[0].Uptime
+		cpuUsage, _ := netgear.StringPercentToFloat(deviceInfo.DeviceInfo.Cpu[0].Usage)
+		memUsage, _ := netgear.StringPercentToFloat(deviceInfo.DeviceInfo.Memory[0].Usage)
```

You're still ignoring errors.